python_ta/debug/id_tracker.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any, Optional
+
+
+class IDTracker:
```

Comment:
Add documentation for this class and the initializer. In particular, explain how this class is intended to support both long-term and short-term id tracking, as this is the most important and subtle part of this class.

Add attribute type annotations in the body of the class, not the initializer. This looks like:

```python
class IDTracker:
    """..."""

    id_counter: int
    # etc.
```